File: listing.cpp
#include <iostream>
#include <fstream>
#include <string>

void setConsoleEncoding();
std::ifstream openPrnFile();
std::ofstream createCsvFile();
void convertPrnToCsv(std::ifstream& prnFile, std::ofstream& csvFile);

int main()
{
    setConsoleEncoding();

    std::ifstream prnFile = openPrnFile();
    std::ofstream csvFile = createCsvFile();

    convertPrnToCsv(prnFile, csvFile);

    prnFile.close();
    csvFile.close();

    std::cout << "Конвертация PRN-файла в CSV-файл завершена успешно" << std::endl;

    system("pause");
    return 0;
}

void setConsoleEncoding()
{
    SetConsoleCP(1251);
    SetConsoleOutputCP(1251);
}

std::ifstream openPrnFile()
{
    std::string input;
    std::string extension_prn = ".prn";
    std::cout << "Введите имя файла для чтения (*.prn): ";
    std::cin >> input;
    input += extension_prn;
    std::ifstream prnFile(input);
    if (!prnFile)
    {
        std::cout << "Не удалось открыть PRN-файл" << std::endl;
        exit(1);
    }
    return prnFile;
}

std::ofstream createCsvFile()
{
    std::string output;
    std::string extension_csv = ".csv";
    std::cout << "Введите имя файла для записи (*.csv): ";
    std::cin >> output;
    output += extension_csv;
    std::ofstream csvFile(output);
    if (!csvFile)
    {
        std::cout << "Не удалось создать CSV-файл" << std::endl;
        exit(1);
    }
    return csvFile;
}

void convertPrnToCsv(std::ifstream& prnFile, std::ofstream& csvFile)
{
    std::string line;
    while (std::getline(prnFile, line))
    {
        std::string token;
        size_t pos = 0;
        while ((pos = line.find_first_of(" \t\n", pos)) != std::string::npos)
        {
            token = line.substr(0, pos);
            csvFile << token << ",";
            line.erase(0, pos + 1);
        }
        csvFile << std::endl;
    }
}

File: listing.cs
using System;
using System.IO;

class Program
{
    static void Main()
    {
        SetConsoleEncoding();

        FileStream prnFile = OpenPrnFile();
        FileStream csvFile = CreateCsvFile();

        ConvertPrnToCsv(prnFile, csvFile);

        prnFile.Close();
        csvFile.Close();

        Console.WriteLine("Конвертация PRN-файла в CSV-файл завершена успешно");
        Console.ReadLine();
    }

    static void SetConsoleEncoding()
    {
        Console.InputEncoding = System.Text.Encoding.GetEncoding(1251);
        Console.OutputEncoding = System.Text.Encoding.GetEncoding(1251);
    }

    static FileStream OpenPrnFile()
    {
        Console.Write("Введите имя файла для чтения (*.prn): ");
        string input = Console.ReadLine();
        string path = input + ".prn";
        
        try
        {
            return new FileStream(path, FileMode.Open, FileAccess.Read);
        }
        catch (FileNotFoundException)
        {
            Console.WriteLine("Не удалось открыть PRN-файл");
            Environment.Exit(1);
            return null; // Эта строка никогда не выполнится, но компилятор требует ее наличие
        }
    }

    static FileStream CreateCsvFile()
    {
        Console.Write("Введите имя файла для записи (*.csv): ");
        string output = Console.ReadLine();
        string path = output + ".csv";

        try
        {
            return new FileStream(path, FileMode.Create, FileAccess.Write);
        }
        catch (IOException)
        {
            Console.WriteLine("Не удалось создать CSV-файл");
            Environment.Exit(1);
            return null; // Эта строка никогда не выполнится, но компилятор требует ее наличие
        }
    }

    static void ConvertPrnToCsv(FileStream prnFile, FileStream csvFile)
    {
        StreamReader reader = new StreamReader(prnFile);
        StreamWriter writer = new StreamWriter(csvFile);

        string line;
        while ((line = reader.ReadLine()) != null)
        {
            string[] tokens = line.Split(new char[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
            foreach (string token in tokens)
            {
                writer.Write(token + ",");
            }
            writer.WriteLine();
        }

        reader.Close();
        writer.Close();
    }
}
